src/vue/modules/verification/general-form/index.vue

```diff
@@ -0,0 +1,206 @@
+  submit: 'submit',
+}
+
+export default {
+  name: 'verification-general-form',
+  components: {
+    SectionCountry,
+    SectionPersonal,
+    SectionAddress,
+    SectionDocuments,
+    SectionSelfie,
+    SectionAvatar,
+  },
+  mixins: [FormMixin],
+  props: {
+    /**
+     * @property config - the config for component to use
+     * @property config.horizonURL - the url of horizon server (without version)
+     */
+    config: { type: Object, required: true },
+    wallet: { type: Wallet, required: true },
+
+    blobId: { type: String, default: '' },
+    requestId: { type: String, required: true },
+
+    generalRoleId: { type: String, required: true },
+    usVerifiedRoleId: { type: String, required: true },
+    usAccreditedRoleId: { type: String, required: true },
+  },
+  data: _ => ({
+    isInitialized: false,
+    isCountryChangeDisabled: false,
+  }),
+  computed: {
+    ...mapGetters('verification-general-form', {
+      isAccredited: types.isAccredited,
+      countryCode: types.countryCode,
+      blobData: types.blobData,
+      country: types.country,
+    }),
+    verificationCode () {
+      return this.wallet.accountId.slice(1, 6)
+    },
+    isUSResident () {
+      return isUSResidence(this.countryCode)
+    },
+    accountRoleToSet () {
+      if (this.isUSResident) {
+        return this.isAccredited
+          ? this.usAccreditedRoleId
+          : this.usVerifiedRoleId
+      }
+
+      return this.generalRoleId
+    },
+  },
+  async created () {
+    initApi(this.wallet, this.config)
+
+    if (this.blobId) {
+      try {
+        this.populateForm(await this.getBlobData(this.blobId))
+      } catch (e) {
+        // catching error to be able to
+        // re-submit incompatible requests
+        log.error(e)
+      }
+
+      // Disabling country change to prevent updating role for
+      // unfulfilled requests. Is temporary until canceling change
+      // role requests is implemented on backend:
+      this.isCountryChangeDisabled = true
+    }
+
+    this.isInitialized = true
+  },
+  methods: {
+    ...mapActions('verification-general-form', {
+      getBlobData: types.GET_BLOB_DATA,
+      populateForm: types.POPULATE_STATE,
+      uploadDocuments: types.UPLOAD_DOCUMENTS,
+      createBlob: types.CREATE_BLOB,
+    }),
+    isAllFormsValid () {
+      let isValid = true
+
+      // we need to trigger errors in all sections,
+      // even if we've already found invalid one
+      Object.values(this.$refs).forEach(ref => {
+        if (!ref.isFormValid()) {
+          isValid = false
+        }
+      })
+
+      return isValid
+    },
+    async submit () {
+      this.disableForm()
+      try {
+        await this.uploadDocuments()
+        const blobId = await this.createBlob(this.wallet.accountId)
+        await this.createRequest(blobId)
+        this.enableForm()
+        this.$emit(EVENTS.submit)
+      } catch (e) {
+        this.enableForm()
+        ErrorHandler.process(e)
+      }
```

Comment:
@max-nes we can't because then race condition will happen. the outer component disables the form by submit event again and can do it (and does) before we enable it here. 

that's why we should enable it exactly before we emit the event

Reply:
Let’s describe it with a comment so